### PR TITLE
consistify main loop across platforms

### PR DESCRIPTION
**Pull request by RemoveRusky**

### CODE REVIEW

**Comment by RemoveRusky** [comment]
this includes fix for #1114

File: ENIGMAsystem/SHELL/Bridges/Win32-Direct3D11/graphics_bridge.cpp [hunk 1]
```diff
@@ -58,7 +58,8 @@ namespace enigma
     enigma_user::draw_clear(enigma_user::window_get_color());
   }
   
-  void EnableDrawing (HGLRC *hRC) {
+  void EnableDrawing(void* handle) {
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Bridges/Win32-Direct3D11/graphics_bridge.cpp]: Why is `void*` necessary here? Bridges can use window-system–specific types. You can always parameterize your main loop logic, or require an `enigma::window_handle_t`.

File: ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp [hunk 2]
```diff
@@ -1,25 +1,112 @@
...
+  return current_time_mcs;
+}
+
+void set_program_args(int argc, char** argv) {
+  parameters = new std::string[argc];
+  parameterc = argc;
+  for (int i = 0; i < argc; i++) parameters[i] = argv[i];
+}
+
+int main(int argc, char** argv, void* windowHandle) {
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp]: Calling this `main` could be a bad idea. Also, at the point where you've added a parameter and moved this to `General`, why not just put it in a header and parameterize it?

File: ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp [hunk 3]
```diff
@@ -1,25 +1,112 @@
...
 
+std::string working_directory = "";
+std::string program_directory = "";
+std::string keyboard_string = "";
+int keyboard_key = 0;
+double fps = 0;
+unsigned long delta_time = 0;
+unsigned long current_time = 0;
+
+std::string parameter_string(int num) { return num < enigma::parameterc ? enigma::parameters[num] : ""; }
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp]: Ahh... This unification is a great side-effect of this change.

File: ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp [hunk 4]
```diff
@@ -1,25 +1,112 @@
...
+  set_program_args(argc, argv);
+
+  initInput();
+
+  if (!initGameWindow()) {
+    printf("Failed to create game window\n");
+    return -4;
+  }
+
+  EnableDrawing(windowHandle);
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Platforms/General/PFmain.cpp]: Perhaps this is another good use case of extension hooks. Enabling window drawing could just be a routine that each bridge adds. It does sound like a trick to integrate that into the configuration, though; due to the nature of being the system's initializer, it wouldn't have an auxiliary initializer to register the hook, so these would have to be listed in the eyaml (or else use a nasty hack to append to a hook list). I can explain if you're interested.

File: ENIGMAsystem/SHELL/Platforms/General/POSIX/Utility.cpp [hunk 5]
```diff
@@ -23,14 +23,88 @@ void windowsystem_write_exename(char* x) {
...
+#endif
+  enigma_user::working_directory = std::string(buffer);
+}
+
+}  //namespace enigma
 
-int parameter_count() { return enigma::parameterc; }
+namespace enigma_user {
 
 void execute_shell(std::string operation, std::string fname, std::string args) {
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Platforms/General/POSIX/Utility.cpp]: Unrelated: Do we have a bug for how bad these are? These are really bad.

**Comment by RobertBColton** [comment]
@RemoveRusky @JoshDreamland two regressions and counting on this pull request so far 😛

**Pull request merged**